src/components/CreateTargetForm/index.js

```diff
@@ -0,0 +1,121 @@
+};
+const { TARGET } = strings;
+
+const CreateTargetForm = ({ onSubmit, onShowTopics, onHide, topic }) => {
+  const validator = useValidation(createTargetValidations);
+
+  const { values, errors, handleValueChange, handleSubmit, handleBlur } = useForm(
+    {
+      onSubmit,
+      validator,
+      validateOnBlur: true,
+    },
+    [onSubmit],
+  );
+
+  const {
+    handleOnPress,
+    targetError,
+    targetStatus,
+    getTopicsError,
+    getTopicsStatus,
+    topicError,
+  } = useCreateTargetForm(topic, TARGET, handleSubmit);
+
+  const inputProps = useTextInputProps(handleValueChange, handleBlur, values);
+
+  const topicText = () => {
+    if (getTopicsError) return TARGET.topicsError;
+    if (getTopicsStatus === LOADING) return TARGET.loadingTopics;
+    return (topic && topic.label) || TARGET.topicPlaceholder;
+  };
+
+  return (
+    <>
+      <TouchableOpacity style={styles.mainContainer} onPress={onHide} />
+      <KeyboardAwareScrollView keyboardShouldPersistTaps="handled" style={styles.container}>
+        <View style={styles.inputContainer}>
+          <Input
+            label={TARGET.area}
+            testID="area-input"
+            keyboardType="numeric"
+            error={errors[FIELDS.areaLength]}
+            invalid={targetError || errors[FIELDS.areaLength]}
+            short={false}
+            placeholder={TARGET.areaPlaceholder}
+            placeholderTextColor={BLACK}
+            {...inputProps(FIELDS.areaLength)}
+          />
+        </View>
+        <View style={styles.inputContainer}>
+          <Input
+            label={TARGET.titleLabel}
+            testID="target-title-input"
+            error={errors[FIELDS.title]}
+            invalid={targetError || errors[FIELDS.title]}
+            short={false}
+            placeholder={TARGET.titlePlaceholder}
+            placeholderTextColor={BLACK}
+            {...inputProps(FIELDS.title)}
+          />
+        </View>
+        <View>
+          <Text style={[common.inputLabel, common.leftPadding]}>{TARGET.topicLabel}</Text>
+          <TouchableOpacity
+            style={[styles.topic, topicError && common.inputBorderError]}
+            onPress={onShowTopics}>
+            <Text style={[styles.topicInput, Platform.OS === 'android' && fonts.bold]}>
+              {topicText()}
+            </Text>
+          </TouchableOpacity>
+        </View>
+        <ErrorView errors={{ ...errors, targetError, topicError }} />
+        <Button
+          testID="save-target-button"
+          handleOnPress={handleOnPress}
+          additionalStyles={styles.button}
+          title={TARGET.buttonTitle}
+          isLoading={targetStatus === LOADING}
+        />
+      </KeyboardAwareScrollView>
+    </>
+  );
+};
+
+CreateTargetForm.propTypes = {
+  onSubmit: func.isRequired,
+  onShowTopics: func.isRequired,
+  onHide: func.isRequired,
+  topic: oneOfType([
+    shape({
+      id: number,
+      label: string,
+    }),
+    bool,
+  ]).isRequired,
```

Comment:
You can have a separate ```shapes.js``` file in constants and define your types there. Otherwise you will probably repeat this shape in every component that receives a topic and if it changes format you will have more work to do.